### PR TITLE
Refactor/common/rename core

### LINKED ISSUE #2604: Rename common/engine to common/core
The common/engine folder actually contains what we are calling common/core -- so we should rename it.

_Originally posted by @mcdurdin in https://github.com/keymanapp/keyman/pull/2603/review_comment/create_

### PR DESCRIPTION
**Pull request by darcywong00**
Fixes #2604 

Before we refactor Keyman Core into the other desktop platforms for 14.0, it will be good to get this out of the way.

TODO: ~update CI accordingly~

### CODE REVIEW

**Comment by darcywong00** [comment]
@ermshiperete , will [KeymanPackagingJobs.groovy](https://github.com/sillsdev/ci-builder-scripts/blob/master/groovy/KeymanPackagingJobs.groovy) be able

File: common/core/keyboardprocessor/meson.build [hunk 1]
```diff
@@ -6,7 +6,7 @@
 #
 
 project('keyboardprocessor', 'cpp', 'c',
-        version: '11.0.0',
+        version: '14.0.17',
```
Comment (ermshiperete) [inline common/core/keyboardprocessor/meson.build]: Too bad we can't get this version number from `VERSION.md`

Comment (darcywong00) [reply common/core/keyboardprocessor/meson.build]: Ah this was inadvertently included. iirc, linux/scripts/[reconf.sh](https://github.com/keymanapp/keyman/blob/master/linux/scripts/reconf.sh#L59) parses version and replaces it in meson.build

Comment (mcdurdin) [reply common/core/keyboardprocessor/meson.build]: Ideally we have a meson.in -> meson.build so meson.build is not included in CI

**Comment by mcdurdin** [comment]
So we should be under common/core/desktop (we'll also ultimately have common/core/web). Rename keyboardprocessor to desktop I think.

**Comment by darcywong00** [comment]
> So we should be under common/core/desktop (we'll also ultimately have common/core/web). Rename keyboardprocessor to desktop

k. I've renamed common/core/keyboardprocessor to common/core/desktop

But internally for Linux, it still copies into `linux/keyboardprocessor` when building because  I didn't want to perturb the package names.

**Comment by mcdurdin** [comment]
I'd like to hold off merging this until #1720 is merged -- they shouldn't impact each other but they are both big changes!

**Comment by mcdurdin** [comment]
> TODO: update CI accordingly

Has this been completed?

File: common/core/desktop/meson.build [hunk 2]
```diff
@@ -6,7 +6,7 @@
 #
 
 project('keyboardprocessor', 'cpp', 'c',
-        version: '11.0.0',
+        version: '14.0.17',
```
Comment (mcdurdin) [inline common/core/desktop/meson.build]: Can we avoid rewriting this file each time there is a version update? Can it be a meson.in -> meson.build or something like that?

**Comment by darcywong00** [comment]
> Has this been completed?

I believe I still need to update the Jenkins script to handle either path. TC is updated

**Comment by ermshiperete** [comment]
> I believe I still need to update the Jenkins script to handle either path. TC is updated

I updated the Jenkins pipeline scripts, but I still have to retire the non-pipeline jobs.

File: common/core/desktop/meson.build (hunk 2 above)
Comment (darcywong00) [reply common/core/desktop/meson.build]: I found the incantation to cat the VERSION.md file.

File: common/core/desktop/meson.build [hunk 3]
```diff
@@ -6,11 +6,15 @@
 #
 
 project('keyboardprocessor', 'cpp', 'c',
-        version: '14.0.17',
+        # Get version number from file.
+        # Fallback to "more" for Windows compatbility
+        version: run_command(find_program('cat', 'more'),
```
Comment (ermshiperete) [inline common/core/desktop/meson.build]: Could use `more` on both Windows and Linux, but this works just as well :smile:

File: common/core/desktop/meson.build [hunk 4]
```diff
@@ -6,11 +6,15 @@
 #
 
 project('keyboardprocessor', 'cpp', 'c',
-        version: '11.0.0',
+        # Get version number from file.
+        # Fallback to "more" for Windows compatbility
+        version: run_command(find_program('cat', 'more'),
+                 files('../../../VERSION.md')).stdout().strip(),
```
Comment (ermshiperete) [inline common/core/desktop/meson.build]: :+1: nice find

File: .gitignore [hunk 5]
```diff
@@ -240,6 +240,7 @@
 
 # VS Code
 .vscode
+.vs
```
Comment (ermshiperete) [inline .gitignore]: `.vs` (as well as `.vscode`) is a directory, isn't it? Then it would be better to write it as `.vs/`

**Pull request merged**